src/codeu/chat/client/simplegui/ChatGuiFX.java

```diff
@@ -138,21 +179,42 @@ public void start(Stage primaryStage) throws Exception {
         chatVBox.setMaxWidth(Double.MAX_VALUE);
         convosVBox.setMaxWidth(150);
 
-        hboxInput.getChildren().addAll(input, send, update);
+        hboxInput.getChildren().addAll(input, sendButton, updateButton);
         userVBox.getChildren().addAll(userTf, users);
         chatVBox.getChildren().addAll(chatTf, messages, hboxInput);
-        convosVBox.getChildren().addAll(convosTf, conversations, addConvo);
+        convosVBox.getChildren().addAll(convosTf, conversations, addConvoButton);
         hboxClient.getChildren().addAll(userVBox, chatVBox, convosVBox);
         container.setCenter(hboxClient);
-        // container.setTop(menuBar);
 
-        mainScene = new Scene(container, WIDTH, HEIGHT);
+        mainScene = new Scene(container, WINDOW_WIDTH, WINDOW_HEIGHT);
         thestage.setScene(signInScene);
         thestage.show();
     }
 
-    public void ButtonClicked(ActionEvent e)
-    {
-        thestage.setScene(mainScene);
+    private void buttonClicked(ActionEvent e) {
+        thestage.setScene(mainScene);           // TODO: Call a controller function here instead
+    }
+
+    private void addConversation(ActionEvent e) {
+        // popup for the user to add a conversation
+        TextInputDialog dialog = new TextInputDialog("Name your conversation!");
+        dialog.setTitle(" "); // make the title blank
+        dialog.setHeaderText("Create your conversation");
+
+        // get input and add the name of the convo
+        String name = dialog.showAndWait().get();
+
+        // TODO: check for duplicate conversations & handle them
+
+        if (name != null && name.length() > 0) {
```

Comment:
Is this all the checking that needs to be done for name? What happens if I try to add bad characters?